src/misc.jl

```diff
@@ -60,7 +60,7 @@ run lengths.
 function inverse_rle(vals::AbstractVector{T}, lens::AbstractVector{<:Integer}) where T
     m = length(vals)
     length(lens) == m || raise_dimerror()
-
+    all(>=(0), lens) || throw(ArgumentError("lengths must be non-negative"))
```

Comment:
Is there a specific reason to support `0`? I would have assumed that a valid run-length encoding does not contain values with length `0`.
```suggestion
    all(>(0), lens) || throw(ArgumentError("lengths must be positive"))
```

I'm also a bit surprised that tests do not error on Julia 1.0 since
```julia
help?> >=(0)
  >=(x)

  Create a function that compares its argument to x using >=, i.e. a function equivalent to y -> y >= x. The returned function is of type
  Base.Fix2{typeof(>=)}, which can be used to implement specialized methods.

  │ Julia 1.2
  │
  │  This functionality requires at least Julia 1.2.

help?> >(0)
  >(x)

  Create a function that compares its argument to x using >, i.e. a function equivalent to y -> y > x. The returned function is of type
  Base.Fix2{typeof(>)}, which can be used to implement specialized methods.

  │ Julia 1.2
  │
  │  This functionality requires at least Julia 1.2.
````

Reply:
> Is there a specific reason to support 0?

Not really but it doesn't hurt to support IMO and may possibly be convenient to someone at some point ¯\\\_(ツ)\_/¯

> I'm also a bit surprised that tests do not error on Julia 1.0 since [...]

Huh yeah, me too. Maybe it existed in 1.0 but was undocumented and/or didn't return a `Fix2`? I don't have 1.0 handy to check.

Reply:
> Is there a specific reason to support `0`? I would have assumed that a valid run-length encoding does not contain values with length `0`.

I can definitely see how someone would want to programmatically create lists where some values can be skipped, e.g.
```julia
julia> r = 3
julia> startidx = [2,5,3]
julia> inverse_rle.(Ref(["A", "B", "C"]), ([i-1, r, max(0,8-i-r)] for i in startidx))
3-element Vector{Vector{String}}:
 ["A", "B", "B", "B", "C", "C", "C"]
 ["A", "A", "A", "A", "B", "B", "B"]
 ["A", "A", "B", "B", "B", "C", "C"]
```

Reply:
I agree allowing zeros can be useful.

I've checked Julia 1.0, `>(0)` and `>=(0)` work thanks to Compat.jl.